servers/rendering/rasterizer_rd/shaders/scene_high_end.glsl

```diff
@@ -1511,371 +1023,896 @@ void sdfgi_process(uint cascade, vec3 cascade_pos, vec3 cam_pos, vec3 cam_normal
+	}
+	*/
+#ifdef LIGHT_TRANSMITTANCE_USED
+	float transmittance_z = transmittance_depth;
+#else
+	float transmittance_z = 0.0;
 #endif
 
-#if defined(LIGHT_ANISOTROPY_USED)
-
-	if (anisotropy > 0.01) {
-		//rotation matrix
-		mat3 rot = mat3(tangent, binormal, normal);
-		//make local to space
-		tangent = normalize(rot * vec3(anisotropy_flow.x, anisotropy_flow.y, 0.0));
-		binormal = normalize(rot * vec3(-anisotropy_flow.y, anisotropy_flow.x, 0.0));
+	float shadow;
+	vec3 shadow_color_enabled = GET_LIGHT_SHADOW_COLOR(idx);
+	if (SPOT_SHADOW_PROCESS(idx, vertex, normal_interp, shadow, transmittance_z)) {
+        vec3 no_shadow = SPOT_PROJECTOR_PROCESS(idx, vertex, vertex_ddx, vertex_ddy);
+		shadow_attenuation = mix(shadow_color_enabled.rgb, no_shadow, shadow);
 	}
 
+	light_compute(normal, normalize(light_rel_vec), eye_vec, size_A, color_specular.rgb, light_attenuation, shadow_attenuation, albedo, roughness, metallic, specular, color_specular.a * p_blob_intensity,
+#ifdef LIGHT_BACKLIGHT_USED
+			backlight,
 #endif
-
-#ifdef ENABLE_CLIP_ALPHA
-	if (albedo.a < 0.99) {
-		//used for doublepass and shadowmapping
-		discard;
-	}
+#ifdef LIGHT_TRANSMITTANCE_USED
+			transmittance_color,
+			transmittance_depth,
+			transmittance_curve,
+			transmittance_boost,
+			transmittance_z,
 #endif
-	/////////////////////// DECALS ////////////////////////////////
-
-#ifndef MODE_RENDER_DEPTH
+#ifdef LIGHT_RIM_USED
+			rim * spot_attenuation, rim_tint,
+#endif
+#ifdef LIGHT_CLEARCOAT_USED
+			clearcoat, clearcoat_gloss,
+#endif
+#ifdef LIGHT_ANISOTROPY_USED
+			binormal, tangent, anisotropy,
+#endif
+#ifdef USE_SHADOW_TO_OPACITY
+			alpha,
+#endif
+			diffuse_light, specular_light);
+}
 
-	uvec4 cluster_cell = texture(usampler3D(cluster_texture, material_samplers[SAMPLER_NEAREST_CLAMP]), vec3(screen_uv, (abs(vertex.z) - scene_data.z_near) / (scene_data.z_far - scene_data.z_near)));
-	//used for interpolating anything cluster related
-	vec3 vertex_ddx = dFdx(vertex);
-	vec3 vertex_ddy = dFdy(vertex);
+uint SPOT_LIGHT_COUNT(uvec4 cluster_cell) {
+	return cluster_cell.y >> CLUSTER_COUNTER_SHIFT;
+}
+uint GET_SPOT_LIGHT(uvec4 cluster_cell, uint index) {
+	uint spot_light_pointer = cluster_cell.y & CLUSTER_POINTER_MASK;
+	uint light_index = cluster_data.indices[spot_light_pointer + index];
+	return light_index;
+}
+uint OMNI_LIGHT_COUNT(uvec4 cluster_cell) {
+	return cluster_cell.x >> CLUSTER_COUNTER_SHIFT;
+}
+uint GET_OMNI_LIGHT(uvec4 cluster_cell, uint index) {
+	uint omni_light_pointer = cluster_cell.x & CLUSTER_POINTER_MASK;
+	uint light_index = cluster_data.indices[omni_light_pointer + index];
+	return light_index;
+}
+uint DIRECTIONAL_LIGHT_COUNT() {
+	return scene_data.directional_light_count;
+}
+uint GET_DIRECTIONAL_LIGHT(uint index) {
+	return index;
+}
+uint REFLECTION_PROBE_COUNT(uvec4 cluster_cell) {
+	return cluster_cell.z >> CLUSTER_COUNTER_SHIFT;
+}
 
-	{ // process decals
+#else // !defined(MODE_RENDER_DEPTH) && !defined(MODE_UNSHADED)
```

Comment:
Is this accurate?

Reply:
Same. Not sure if this is referring to the green or the red, but the diff here is comparing two unrelated parts of code.

Reply:
Remove the comment if inaccurate?